src/common-principles.md

```diff
@@ -436,6 +393,53 @@ In particular, if a BIDS dataset contains a `derivatives/` subdirectory,
 the contents of that directory may be a heterogeneous mix of BIDS Derivatives
 datasets and non-compliant derivatives.
 
+## Project dataset
+
+BIDS allows one to organize the data for the entire project (original source data, raw BIDS, derivatives) as a valid BIDS dataset in the following way
+
+<!-- This block generates a file tree.
+A guide for using macros can be found at
+ https://github.com/bids-standard/bids-specification/blob/master/macros_doc.md
+-->
+{{ MACROS___make_filetree_example(
+    {
+    "my_project-1": {
+        "sourcedata": {
+            "dicoms": {},
+            "raw": {
+                "sub-01": {},
+                "sub-02": {},
+                "...": "",
+                "dataset_description.json": "",
+				"...": "",
+            },
+            "..." : "",
+        },
+        "derivatives": {
+            "pipeline_1": {},
+            "pipeline_2": {},
+            "...": "",
+        },
+        "dataset_description.json": "",
+        "...": "",
+    }
+   }
+) }}
+
+In this example, `sourcedata/dicoms` is not nested inside
+`sourcedata/raw`, **and only the `sourcedata/raw` subdirectory** is a BIDS-compliant dataset among `sourcedata/` subfolders.
+The subdirectories of `derivatives` MAY be BIDS-compliant derivatives datasets
+(see [Non-compliant derivatives](#non-compliant-derivatives) for further discussion).
+The above example is a fully compliant BIDS dataset, providing a convention useful for organizing source, raw BIDS, and derived BIDS data while maintaining overall BIDS compliance.
+When using this convention, `dataset_description.json` MUST have `DatasetType` to be set to `"project"`.  It is also RECOMMENDED to set the `SourceDatasets`
+field in `dataset_description.json` of each subdirectory of `derivatives` to:
+
+```JSON
+{
+  "SourceDatasets": [ {"URL": "../../sourcedata/raw/"} ]
```

Comment:
```suggestion
  "SourceDatasets": [ {"raw": "../../sourcedata/raw/"} ]
```

The dataset name "URL" doesn't really make sense to me.

Reply:
I think it is ok (note that I am just moving this piece of doc). It is defined in https://github.com/bids-standard/bids-specification/blob/HEAD/src/schema/objects/metadata.yaml#L3170C1-L3190C21 .

But your point is good that I thought that we might want to add explicit `Path` to that schema for those cases where source dataset is included within current dataset e.g. somewhere under `sourcedata/`.  This would harmonize with e.g. `.gitmodules` which has both URL and path, and they could even be the same. But decided not to bother raising that point ATM.  WDYT -- worth discussing further?

Reply:
I will resolve this one, but if you feel it would benefit from more discussion, feel welcome to unresolve @tsalo !